messaging/src/main/java/org/axonframework/common/caching/EhCacheAdapter.java

```diff
@@ -16,169 +16,168 @@
 
 package org.axonframework.common.caching;
 
-import net.sf.ehcache.CacheException;
-import net.sf.ehcache.Ehcache;
-import net.sf.ehcache.Element;
-import net.sf.ehcache.event.CacheEventListener;
 import org.axonframework.common.Registration;
-
+import org.ehcache.core.Ehcache;
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventFiring;
+import org.ehcache.event.EventOrdering;
+import org.ehcache.event.EventType;
+
+import java.util.EnumSet;
 import java.util.function.UnaryOperator;
 
 /**
  * Cache implementation that delegates all calls to an EhCache instance.
  *
  * @author Allard Buijze
+ * @author Gerard Klijs
  * @since 2.1.2
- * @deprecated since ehcache 2 is nearing eol, we added {@link EhCache3Adapter} instead, which uses ehcache 3.
  */
-@Deprecated
+@SuppressWarnings("rawtypes")
 public class EhCacheAdapter extends AbstractCacheAdapter<CacheEventListener> {
```

Comment:
As this rename is a breaking change, we might want to a rewrite rule fo this?

Reply:
Completely agree, we should have something for this.
For now, I'll add it to the `axon5/api-changes.md`, which we can later use to construct the Rewrite rules. 
Or, that was my idea to resolve this. What do you think?